### PR TITLE
SI-6113 typeOf now works for type lambdas

### PR DESCRIPTION
**Pull request by xeno-by**
It's not a problem to have an abstract type symbol in a ground type tag
if that symbol is defined in the type being reified.

This mechanics was already built in for existentials, now we extend it to
include type params of poly types.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @paulp @retronym

**Comment by xeno-by** [comment]
@adriaanm That's one of the bugs, which Mark mentioned would be nice to have in 2.10.1. The fix is trivial, so I decided to submit it even on the verge of the deadline.

File: src/compiler/scala/reflect/reify/phases/Reify.scala [hunk 1]
```diff
@@ -28,7 +28,10 @@ trait Reify extends GenSymbols
       finally currents = currents.tail
     }
   }
-  def currentQuantified = flatCollect(reifyStack.currents)({ case ExistentialType(quantified, _) => quantified })
+  def boundSymbolsInCallstack = flatCollect(reifyStack.currents)({
+    case ExistentialType(quantified, _) => quantified
+    case PolyType(typeParams, _) => typeParams
+  })
```
Comment (retronym) [inline src/compiler/scala/reflect/reify/phases/Reify.scala]: Lose the redundant parens, and it LGTM.

Comment (xeno-by) [inline src/compiler/scala/reflect/reify/phases/Reify.scala]: Done

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by adriaanm** [comment]
"pending OK" means it's a success waiting to go green once the other job goes "success"
that's the bug I was referring to, the kitten doesn't realize the other job goes green
it can't just put in "success" because then the job would look successful as soon as the first job finishes,
while you have to wait for all jobs to finish

i'll look into fixing the kitten this weekend, but it's not critical: it's running the builds, the label is added correctly, the buildlog is correct as well and it's overly conservative, no false positives

to double check, you can click on the links in the build log and see that the build was ok

**Comment by adriaanm** [comment]
PLS SYNCH

**Comment by adriaanm** [comment]
BUILDLOG?

**Comment by JamesIry** [comment]
LGTM when the kitty gets its act together.

**Pull request merged**